FnCAS/test.cc

```diff
@@ -67,63 +68,44 @@ static_assert(std::is_same<fncas::term_vector_t, std::vector<fncas::term_t>>::va
 // It's derived from the now-custom `std::vector<fncas::term_t>`.
 static_assert(std::is_base_of<fncas::term_vector_t, fncas::variables_vector_t>::value, "");
 
-static_assert(std::is_same<std::vector<fncas::double_t>, fncas::impl::V2X<fncas::double_t>>::value, "");
-static_assert(std::is_same<fncas::impl::X2V<std::vector<fncas::double_t>>, fncas::double_t>::value, "");
-
-static_assert(std::is_same<fncas::impl::V, fncas::impl::X2V<fncas::impl::X>>::value, "");
-static_assert(std::is_same<fncas::impl::X, fncas::impl::V2X<fncas::impl::V>>::value, "");
-
-TEST(FnCAS, FNCAS_JIT_VALUE) {
-  const auto CLANG = []() { return "CLANG"; };
-  const auto AS = []() { return "AS"; };
-  const auto NASM = []() { return "NASM"; };
-  static_cast<void>(CLANG);
-  static_cast<void>(AS);
-  static_cast<void>(NASM);
-#ifdef FNCAS_JIT
-  std::cout << "#define FNCAS_JIT " << FNCAS_JIT() << std::endl;
-#else
-  std::cout << "#undef FNCAS_JIT" << std::endl;
-#endif
-}
-
 TEST(FnCAS, ReallyNativeComputationJustToBeSure) {
   EXPECT_EQ(25, SimpleFunction(std::vector<fncas::double_t>({1, 2})));
 }
 
 TEST(FnCAS, NativeWrapper) {
-  fncas::function_reference_t fn(SimpleFunction<fncas::double_t>, 2);
+  fncas::function_t<fncas::JIT::NativeWrapper> fn(SimpleFunction<fncas::double_t>, 2);
   EXPECT_EQ(25.0, fn({1.0, 2.0}));
 }
 
 TEST(FnCAS, IntermediateWrapper) {
   fncas::variables_vector_t x(2);
-  fncas::function_blueprint_t fi = SimpleFunction(x);
+  fncas::function_t<fncas::JIT::Blueprint> fi = SimpleFunction(x);
   EXPECT_EQ(25.0, fi({1.0, 2.0}));
   EXPECT_EQ("((x[0]+(x[1]*2))*(x[0]+(x[1]*2)))", fi.debug_as_string());
 }
 
-#ifdef FNCAS_JIT
+#ifdef FNCAS_JIT_COMPILED
 // This test is here mostly to have the easily locatable autogenerated files in `/tmp` (on Linux).
 TEST(FnCAS, TrivialCompiledFunctions) {
   {
     // grep for `42.000`.
     const fncas::variables_vector_t x(1);
-    const fncas::function_blueprint_t intermediate_function([](const fncas::term_vector_t& x) { return x[0] + 42; }(x));
+    const fncas::function_t<fncas::JIT::Blueprint> intermediate_function(
+        [](const fncas::term_vector_t& x) { return x[0] + 42; }(x));
     EXPECT_EQ(1042, intermediate_function({1000}));
 
-    const fncas::function_compiled_t compiled_function(intermediate_function);
+    const fncas::function_t<fncas::JIT::AS> compiled_function(intermediate_function);
```

Comment:
Shouldn't we use `JIT::Default` everywhere, except the tests specific for the particular JIT type?